src/meta/src/barrier/mod.rs

```diff
@@ -592,6 +597,7 @@ where
                 .update_inflight_prev_epoch(self.env.meta_store())
                 .await
                 .unwrap();
+            checkpoint = checkpoint || checkpoint_control.try_get_checkpoint();
```

Comment:
```suggestion
            let checkpoint = checkpoint || checkpoint_control.try_get_checkpoint();
```

Reply:
I think it should be the `ScheduledBarriers`'s responsibility to decide whether a plain barrier should be a checkpoint, as the periodic barrier is generated by it. IIUC, `CheckpointControl` should be used to manage the **in-flight** barriers.